File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> positions;

        for (int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if (positions.count(diff) != 0) {
                return {positions[diff], i};
            } else {
                positions[nums[i]] = i;
            }
        }

        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indices = new HashMap<>();
        int[] res = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (indices.containsKey(diff)) {
                res[0] = indices.get(diff);
                res[1] = i;
                return res;
            } else {
                indices.put(nums[i], i);
            }
        }

        return res;
    }
}
